File: C/Qwest04/ex05/my_robot_simulator.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

char *my_robot_simulator(char *a)
{
  int x = 0;
  int y = 0;
  char *dir[4];
  dir[0] = "north";
  dir[1] = "east";
  dir[2] = "south";
  dir[3] = "west";
  int orient = 0; // 0-N, 1-E, 2-S, 3-W
  for (int i = 0; i < strlen(a); i++)
  {
    if (a[i] == 'A')
    {
      if (orient == 0)
        y--;
      if (orient == 1)
        x++;
      if (orient == 2)
        y++;
      if (orient == 3)
        x--;
    }
    if (a[i] == 'R')
    {
      if (orient == 3)
        orient = 0;
      else
        orient++;
    }
    if (a[i] == 'L')
    {
      if (orient == 0)
        orient = 3;
      else
        orient--;
    }
  }
  char *buffer = (char *)malloc(100 * sizeof(char));
  sprintf(buffer, "{x: %d, y: %d, bearing: '%s'}", x, y, dir[orient]);
  return buffer;
}

File: JS/Bootcamp JS/My Robot Simulator/ex00/my_robot_simulator.js
function my_robot_simulator (instruct){
    var direction = 'north';
    var x = 0;
    var y = 0;
    
    /*var rev = {
        x: 0,
        y: 0,
        "bearing": 'north'
    };*/
    for (var i = 0; i < instruct.length; i++){

       if (instruct[i] == 'R' && direction == 'east'){
           direction = 'south';

       }else if (instruct[i] == 'R' && direction == 'south'){
           direction = 'west';

       }
       else if (instruct[i] == 'R' && direction == 'west'){
           direction = 'north';

       } else if (instruct[i] == 'R' && direction == 'north'){
           direction = 'east';

       } else if (instruct[i] == 'L' && direction == 'south'){
           direction = 'east';

       } else if (instruct[i] == 'L' && direction == 'west'){
           direction = 'south';

       } else if (instruct[i] == 'L' && direction == 'north'){
           direction = 'west';
       } else if (instruct[i] == 'L' && direction == 'east'){
           direction = 'north';

       } else if (instruct[i] == 'A' && direction == 'south'){
           y++;

       } else if (instruct[i] == 'A' && direction == 'west'){
           x--;

       } else if (instruct[i] == 'A' && direction == 'north'){
           y--;
        } else if (instruct[i] == 'A' && direction == 'east'){
           x++;
        }
    }
    /*rev ["x"] = x;
    rev ["y"] = y;
    rev ["bearing"] = direction;
    return rev;*/
    var ans = "{x: " +x+", y: "+y+", bearing: '"+direction+"'}";
    return ans;
}